doc/hooks/patch-activation-hooks.section.md

```diff
@@ -0,0 +1,51 @@
+
+# `patchActivationBash`, `patchActivationCsh`, `patchActivationFish`, `patchActivationPosix` {#sec-patchActivationHooks}
+
+These hooks provide utilities to patch
+shell scripts meant to be sourced (e.g. the "activate" scripts).
```

Comment:
Sourced where, by whom? I think this needs a more precise description of context. Please check if this is called "activate" scripts everywhere, use the canonical name, and link to the respective section in the docs. Isn't this primarily about build hooks?

```suggestion
# `patchActivation` hooks {#sec-patchActivationHooks}

These hooks provide utilities to patch shell scripts meant to be sourced (e.g. the "activate" scripts):
```

This should be followed by the list of hooks.

Reply:
The hooks are to patch the scripts meant to be sourced *by the software users*, not the implementation of another setup hook. A setup hook can get all it needs by the `substitutions` attribute and the `@foo@` syntax already, while the target scripts are provided by the upstream and might not be patch-able through string-finding-and-replacing.

> This should be followed by the list of hooks.

There seems to be two places to document the setup hooks inside the Nixpkgs Manual, one is [6.7. Package setup hooks (ssec-setup-hooks)](https://nixos.org/manual/nixpkgs/stable/#ssec-setup-hooks), the other is [Chapter 16. Hooks reference (chap-hooks)](https://nixos.org/manual/nixpkgs/stable/#chap-hooks). Not sure which one to go for.